### PR TITLE
Handle failed redirects when getting source URL 

### PR DESCRIPTION
**Pull request by MonicaGrages**
The `get_source_url` method looks for the URL of the first request, unless a redirect happens, in which case it looks for the URL of the second request. In the case of [this run](https://monitoring.rigor.com/checks/146466/runs/6712050383), there was a redirect, but it failed, so there is only one request, and that's causing [this error](https://rollbar.com/rigor/reporting.rigor.com/items/834/). Instead, we should just not return a source URL or domain in this case.

### CODE REVIEW

**Comment by marktisham** [comment]
Hi @mbeasley ! Might want to check the perms on this repo, somehow i got alerted :-). Not that i don't mind keeping in the loop :-). Hope all is going well!

**Comment by robertam3** [comment]
@marktisham hi! This repo is public so we can't unsubscribe for you. You need to click "unwatch" if you no longer want to get notifications:

image

**Comment by marktisham** [comment]
Ah that works, thanks @rcrl3 ! Helpful as ever, hope all is well!

File: lib/har/stats.rb [hunk 1]
```diff
@@ -136,10 +136,12 @@ def calculate_server_time
     end
 
     def get_source_url
-      pages.first.is_redirect? ? entries[1].request.url.slice(0, 255) : entries.first.request.url.slice(0, 255)
+      pages.first.is_redirect? ? entries[1]&.request&.url&.slice(0, 255) : entries.first.request.url.slice(0, 255)
```
Comment (mbeasley) [inline lib/har/stats.rb]: Should you also add safe nav to the other side of the conditional?

**Review by mbeasley — APPROVED** [review]
LGTM, with one question.

File: lib/har/stats.rb (hunk 1 above)
Comment (MonicaGrages) [reply lib/har/stats.rb]: Yeah. I'm not sure if we would get to this point if there were no requests. But I don't think it could hurt.

**Pull request merged**